packages/zoe/src/contracts/multipoolAutoswap/multipoolAutoswap.js

```diff
@@ -80,11 +80,39 @@ const start = zcf => {
   const {
     getOutputForGivenInput,
     getInputForGivenOutput,
-  } = makeGetCurrentPrice(isSecondary, isCentral, getPool);
+    getPriceGivenAvailableInput: getInternalPriceGivenAvailableInput,
+    getPriceGivenRequiredOutput: getInternalPriceGivenRequiredOutput,
+  } = makeGetCurrentPrice(isSecondary, isCentral, getPool, centralBrand);
+
+  const getPriceGivenRequiredOutput = (brandIn, amountOutInitial) => {
+    const { amountIn, amountOut } = getInternalPriceGivenRequiredOutput(
+      brandIn,
+      amountOutInitial,
+    );
+    // dropping centralAmount from the public method
+    return { amountIn, amountOut };
+  };
+
+  const getPriceGivenAvailableInput = (amountInInitial, brandOut) => {
+    const { amountIn, amountOut } = getInternalPriceGivenAvailableInput(
+      amountInInitial,
+      brandOut,
+    );
+    // dropping centralAmount from the public method
+    return { amountIn, amountOut };
+  };
+
   const {
     makeSwapInInvitation,
     makeSwapOutInvitation,
-  } = makeMakeSwapInvitation(zcf, isSecondary, isCentral, getPool);
+  } = makeMakeSwapInvitation(
+    zcf,
+    isSecondary,
+    isCentral,
+    getPool,
+    getInternalPriceGivenAvailableInput,
+    getInternalPriceGivenRequiredOutput,
```

Comment:
Also, we can get rid of passing these new methods in, as I think we should not use them in swap. I think they should only be used as a helpful method if users want to, prior to making their offer.